### PR TITLE
Allow top-level arrays for "application/*json" mime type

### PR DESCRIPTION
**Pull request by gnestor**
Closes https://github.com/jupyter/nbformat/issues/57

### CODE REVIEW

**Comment by takluyver** [comment]
Ping @minrk. I think this may be problematic because some json has been stored double-serialised, and we use arrays to store multi-line strings more neatly.

**Comment by gnestor** [comment]
I assumed that it was a bit more complicated than this. I am testing it locally and it's working fine so far...

File: nbformat/v4/nbformat.v4.schema.json [hunk 1]
```diff
@@ -375,7 +375,7 @@
                 "patternProperties": {
                     "^application/([a-zA-Z0-9.]+\\+)?json$": {
                         "description": "Mimetypes with JSON output, can be an object",
-                        "type": "object"
+                        "type": ["object", "array"]
```
Comment (minrk) [inline nbformat/v4/nbformat.v4.schema.json]: json outputs should be allowed to be any type, not just object and array

**Comment by rgbkrk** [comment]
> I think we can do this in a minor revision

❤️

File: nbformat/v4/nbformat.v4.schema.json (hunk 1 above)
Comment (gnestor) [inline nbformat/v4/nbformat.v4.schema.json]: Just comit the `type` property. See https://github.com/json-schema/json-schema/issues/172#issuecomment-114076650.

**Comment by gnestor** [comment]
@minrk I omitted the `type` property which effectively allows any type and it's working for me locally 👍

File: nbformat/v4/nbformat.v4.schema.json (hunk 1 above)
Comment (rgbkrk) [inline nbformat/v4/nbformat.v4.schema.json]: Even better!

File: nbformat/v4/nbformat.v4.schema.json [hunk 2]
```diff
@@ -374,8 +374,7 @@
                 },
                 "patternProperties": {
                     "^application/([a-zA-Z0-9.]+\\+)?json$": {
-                        "description": "Mimetypes with JSON output, can be an object",
-                        "type": "object"
+                        "description": "Mimetypes with JSON output, can be an object"
```
Comment (minrk) [inline nbformat/v4/nbformat.v4.schema.json]: description needs updating now: `can be any type`

Comment (gnestor) [inline nbformat/v4/nbformat.v4.schema.json]: Done!

**Pull request merged**